{{cookiecutter.project_name}}/contracts/NFT.vy

```diff
@@ -301,6 +317,49 @@ def royaltyInfo(_tokenId: uint256, _salePrice: uint256) -> (address, uint256):
     return self.owner, royalty
 {%- endif %}
 
+{%- if cookiecutter.force_royalties == 'y' %}
+# Helper functions in case market place does not support royalties to execute by this contract with _deductRoyalties()
+@internal
+@view
+def _royaltyInfo(_tokenId: uint256, _salePrice: uint256) -> (address, uint256):
+    """
+    /// @notice Called with the sale price to determine how much royalty
+    //          is owed and to whom. Important; Not all marketplaces respect this, e.g. OpenSea
+    /// @param _tokenId - the NFT asset queried for royalty information
+    /// @param _salePrice - the sale price of the NFT asset specified by _tokenId
+    /// @return receiver - address of who should be sent the royalty payment
+    /// @return owner address and royaltyAmount - the royalty payment amount for _salePrice
+    """
+
+    royalty: uint256 = convert(convert(_salePrice, decimal) * ROYALTY_TO_APPLY_TO_PRICE, uint256) # Percentage that accepts decimals
+    return self.owner, royalty
+
+@internal
+def royaltyChecker(tokenId: uint256):
+     # check if royalties hace been paid
+     if self.balance < self.lastBalance:
+        self._deductRoyalties(tokenId)
+        # equal the contract balance to the lastBalance for future checks
+        self.lastBalance = self.balance
```

Comment:
```suggestion
    assert self.balance > self.minRoyaltyAmount
    # Send all balance to the owner (clears `self.balance`)
    send(self.owner, self.balance)

```

Reply:
@fubuloubu for `assert self.balance >= self.minRoyaltyAmount`to make sense, we need to set it back to 0 or (as suggested on my PR, check for a delta with `lastBalance` since the last change.

Reply:
Also withdrawRoyalties() might be very beneficial to keep from a fiscal perspective so that the artist can control when he gets paid, therefore pay his/her taxes... checking this with a lawyer expert on the token matter.

Reply:
> @fubuloubu for `assert self.balance >= self.minRoyaltyAmount`to make sense, we need to set it back to 0 with the `send(self.owner, self.balance)`→ OK. Would you allow the creator to modify the `minRoyaltyAmount`? I think that could be a nice feature is case is set too hight/low at genesis

That's the trick! It'll automatically be 0 because the balance will be physically transferred via `send`

Reply:
> Also `withdrawRoyalties()` might be very beneficial to keep from a fiscal perspective so that the artist can control when he gets paid, therefore pay his/her taxes... checking this with a lawyer expert on the token matter.

From a tax perspective, income is income the moment it's earned (made available to you under your full control), so I don't think there's any real tax optimization tricks here